scripts/append_metadata.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
```

Comment:
Again here, I have some worries about portability of this script. I guess on CI this may be fine, but if this breaks most people's windows build that would be a problem

Reply:
I went back to... invoking a cmake script, see `scripts/append_metadata.cmake`.

This should properly solve portability.